File: pr-dirtdata-lib/data_for_test_search_tool/go/utils/userdata.go
// utils/userdata.go
package utils

import "fmt"

// UserProfile структура для данных пользователя
type UserProfile struct {
	ID    int
	Name  string
	Level int
	Gold  int
}

// LoadUserProfile загружает профиль пользователя
func LoadUserProfile(userID int) UserProfile {
	fmt.Printf("Загрузка данных для пользователя %d...\n", userID)
	return UserProfile{
		ID:    userID,
		Name:  "Test User",
		Level: 5,
		Gold:  100,
	}
}

// SaveUserProfile сохраняет данные профиля пользователя
func SaveUserProfile(userID int, data map[string]interface{}) bool {
	fmt.Printf("Сохранение данных для пользователя %d: %v\n", userID, data)
	return true
}

File: pr-dirtdata-lib/data_for_test_search_tool/typescript/utils/userData.ts
// utils/userData.ts

interface UserProfile {
    id: number;
    name: string;
    level: number;
    gold: number;
}

export function loadUserProfile(userId: number): UserProfile {
    /**
     * Загружает профиль пользователя по его ID.
     */
    console.log(`Загрузка данных для пользователя ${userId}...`);
    return {
        id: userId,
        name: "Test User",
        level: 5,
        gold: 100
    };
}

export function saveUserProfile(userId: number, data: Partial<UserProfile>): boolean {
    /**
     * Сохраняет данные профиля пользователя.
     */
    console.log(`Сохранение данных для пользователя ${userId}:`, data);
    return true;
}